lib/portlayer/storage/vsphere/vm.go

```diff
@@ -29,18 +29,26 @@ func VolumeJoin(op trace.Operation, handle *exec.Handle, volume *storage.Volume,
 	defer trace.End(trace.Begin("vsphere.VolumeJoin"))
 
 	if _, ok := handle.ExecConfig.Mounts[volume.ID]; ok {
-		return nil, fmt.Errorf("Volume with ID %s is already in container %s's mountspec'", volume.ID, handle.ExecConfig.ID)
+		return nil, fmt.Errorf("Volume with ID %s is already in container %s's mountspec config", volume.ID, handle.ExecConfig.ID)
 	}
 
-	newMountSpec := executor.MountSpec{
-		Source: url.URL{
-			Scheme: "label",
-			Path:   volume.Label,
-		},
-		Path: mountPath,
-		Mode: diskOpts["Mode"],
+	//constuct MountSpec for the tether
+	mountSpec := createMountSpec(volume, mountPath, diskOpts)
+
+	//append a device addition spec change to the container config
+	diskDevice := createVolumeVirtualDisk(volume)
+	config := createDeviceConfigSpec(diskDevice)
```

Comment:
Because these are specific unexported helpers for creating the exact configfuration we need in this instance. They are just meant to tidy up this function and divide out the logical creation steps in a readable manner.